applications/launchpad/gui-react/src/components/TBot/TBotPrompt/MessageBox.tsx

```diff
@@ -58,11 +68,17 @@ const MessageBox = (
       <MessageSpaceContainer>
         <MessageSlideIn style={{ ...useSlideInAnim }}>
           <StyledMessage
+            ref={messageBoxRef}
             style={{ ...useOpacityAnim }}
             $skipButton={skipButton}
             $floating={floating}
           >
-            {children}
+            {React.Children.map(children, child => {
```

Comment:
why is this necessary? I think you should isolate this logic and add a comment explaining why we react clone, cause this is usually pretty specific